internal/clients/vault_secrets_preview.go

```diff
@@ -61,27 +61,41 @@ func OpenVaultSecretsAppSecrets(ctx context.Context, client *Client, loc *shared
 
 	var secrets *secret_service.OpenAppSecretsOK
 	var err error
+	var result []*secretmodels.Secrets20231128OpenSecret
+	nextPage := ""
 	for attempt := 0; attempt < retryCount; attempt++ {
-		secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
-		if err != nil {
-			var serviceErr *secret_service.OpenAppSecretDefault
-			ok := errors.As(err, &serviceErr)
-			if !ok {
+		for {
+			if nextPage != "" {
+				params.PaginationNextPageToken = &nextPage
+			}
+			secrets, err = client.VaultSecretsPreview.OpenAppSecrets(params, nil)
+			if err != nil {
+				var serviceErr *secret_service.OpenAppSecretDefault
+				ok := errors.As(err, &serviceErr)
+				if !ok {
+					return nil, err
+				}
+				if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
+					continue
+				}
 				return nil, err
 			}
-			if shouldRetryWithSleep(ctx, serviceErr, attempt, []int{http.StatusTooManyRequests}) {
-				continue
+			result = append(result, secrets.GetPayload().Secrets...)
+			pagination := secrets.GetPayload().Pagination
+			if pagination == nil || pagination.NextPageToken == "" {
+				return result, nil
 			}
-			return nil, err
+
+			nextPage = pagination.NextPageToken
+			break
```

Comment:
I believe this `break` shouldn't be here

Reply:
Yup, meant to make the retry loop as inner loop but forgot to :) Thanks for pointing it out!